b2sdk/transfer/emerge/emerger.py

```diff
@@ -130,14 +163,26 @@ def emerge_stream(
 
         :param int min_part_size: lower limit of part size for the transfer planner, in bytes
         :param int max_part_size: upper limit of part size for the transfer planner, in bytes
-
+        :param str,None large_file_sha1: result file SHA1 hash or ``None`` if not known
         """
         planner = self.get_emerge_planner(
             min_part_size=min_part_size,
             recommended_upload_part_size=recommended_upload_part_size,
             max_part_size=max_part_size,
         )
+
+        # ensure that write_intent_iterator is an iterator (and not just iterable, like a list)
```

Comment:
You can use `iter(iterator)` on both iterator and iterable. It creates a copy if that's an iterator and an iterator if that's an iterable. So no need for a check.